python/tank/util/shotgun/publish_util.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Utility methods related to Published Files in Shotgun
+"""
+
+from __future__ import with_statement
+
+from ...log import LogManager
+from ..shotgun_path import ShotgunPath
+from .. import constants
+from .. import login
+
+log = LogManager.get_logger(__name__)
+
+# global variables
+g_entity_display_name_lookup = None
```

Comment:
Maybe we could introduce a tk.globals ? This ideal smells however. Not that I've written it I'm not sure I like it either.